plenoptic/tools/conv.py

```diff
@@ -1,143 +1,132 @@
 import numpy as np
 import torch
 from torch import Tensor
-from torch import nn
 import torch.nn.functional as F
-import pyrtools as pt
 from typing import Union, Tuple
 import math
 
 
-# TODO under development
-# needs documentation and testing
-# handle multiple channels
-# handle batch dimension
-# handle signal of dimension 1,2,3
-# faster implementation with separable 1d conv
+def correlate_downsample(image, filt, padding_mode="reflect"):
+    """Correlate with a filter and downsample by 2
 
-
-def correlate_downsample(signal, filt, edges="reflect1",
-                         step=2, start=(0, 0), stop=None):
-    """compute the correlation of `signal` with `filter`
-
-    Args:
-        signal ([type]): [description]
-        filt ([type]): [description]
-        edges (str, optional): [description]. Defaults to "reflect1".
-        step (int, optional): [description]. Defaults to 2.
-        start (tuple, optional): [description]. Defaults to (0, 0).
-        stop ([type], optional): [description]. Defaults to None.
-
-    Returns:
-        [type]: [description]
+    Parameters
+    ----------
+    image: torch.Tensor of shape (batch, channel, height, width)
+        Image, or batch of images. Channels are also treated as batches.
+    filt: 2-D torch.Tensor
+        The filter to correlate with the input image
+    padding_mode: string, optional
+        One of "constant", "reflect", "replicate", "circular" or "zero" (same as "constant")
```

Comment:
why have `"zero"` as an option if it's redundant with `"constant"`?

Reply:
When the function `F.pad` takes the argument `mode="constant"` without specifying the value of the constant, the padded constant is default to 0. I added the option of `"zero"` because I thought it was clearer than `"constant"`, but still kept the `"constant"` option.

Reply:
So `"zero"` must be the same `"constant"` here because we can't pass any additional arguments to `F.pad` via `same_padding`.

I'm not a big fan of this -- since we're basically passing the arguments directly to `F.pad`, I think we should keep the arguments the same. So I'd prefer we remove `"zero"` and note that `"constant"` will zero-pad. I think we should be consistent with pytorch, basically.

Reply:
Ok, will do.